src/plugins_3rdparty/primer3/src/Primer3Dialog.cpp

```diff
@@ -657,9 +658,12 @@ bool Primer3Dialog::doDataExchange() {
             }
             qualityList.append(value);
         }
-        if (!qualityList.isEmpty() && (qualityList.size() != (rs->getRegion().length))) { // todo add check on wrong region
+        int qualityLength = qualityList.size();
+        int sequenceLength = context->getSequenceLength();
```

Comment:
nit: this should be qint64 or the whole dialog should be protected to never be called for sequences with length > MAX_INT